yaksh/code_evaluator.py

```diff
@@ -97,45 +113,55 @@ def setup(self):
         if self.in_dir:
             if not os.path.exists(self.in_dir):
                 os.makedirs(self.in_dir)
-        self._change_dir(self.in_dir)
 
-    def safe_evaluate(self, user_answer, partial_grading, test_case_data, file_paths=None):
+    def get_evaluator_objects(self, kwargs):
+        metadata = kwargs.get('metadata')
+        test_case_data = kwargs.get('test_case_data')
```

Comment:
Why are you keeping two different things, metadata and test_case_data?  Why not just have them all as kwargs and send them as such?  What does this split bring you?